builder/builder-source-patch.c

```diff
@@ -254,6 +273,51 @@ builder_source_patch_extract (BuilderSource  *source,
   return TRUE;
 }
 
+static gboolean
+builder_source_patch_bundle (BuilderSource  *source,
+                             BuilderContext *context,
+                             GError        **error)
+{
+  BuilderSourcePatch *self = BUILDER_SOURCE_PATCH (source);
+
+  g_autoptr(GFile) src = NULL;
+  g_autoptr(GFile) patches_dir = NULL;
+  g_autoptr(GFile) destination_file = NULL;
+  g_autofree char *patches_dir_path = NULL;
+  g_autofree char *file_name = NULL;
+  g_autofree char *destination_file_path = NULL;
+  g_autofree char *app_dir_path = g_file_get_path (builder_context_get_app_dir (context));
```

Comment:
Nitpick: you can initialize `app_dir_path` right before `patches_dir_path` instead